### PR TITLE
Always sync text changes and active doc info to remote workspace

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Build: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=10546686&view=results

### CODE REVIEW

**Comment by ToddGrun** [comment]
> Build: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=10541569&view=results

Hmm, that didn't have the desired effect. Lots of TagSpan<IContainerStructureTag>[] allocations in the test insertion.

File: src/EditorFeatures/Core/Remote/SolutionChecksumUpdater.cs [hunk 1]
```diff
@@ -155,6 +151,14 @@ private void OnWorkspaceChanged(object? sender, WorkspaceChangeEventArgs e)
...
 
+        // Check if we're currently paused.  If so ignore this notification.  We don't want to any work in response
+        // to whatever the workspace is doing.
+        lock (_gate)
+        {
+            if (_isSynchronizeWorkspacePaused)
+                return;
+        }
+
         _synchronizeWorkspaceQueue.AddWork();
```
Comment (ToddGrun) [inline src/EditorFeatures/Core/Remote/SolutionChecksumUpdater.cs:162]: >_synchronizeWorkspaceQueue [](http://example.com/codeflow?start=8&length=26)

This didn't change so I'm sure it's fine, but the Resume/Pause methods execute the calls onto the _synchronizeWorkspaceQueue from within the lock.

File: src/Workspaces/Core/Portable/Workspace/Solution/Solution_SemanticModelCaching.cs [hunk 2]
```diff
@@ -2,58 +2,50 @@
...
-    /// Strongly held reference to the semantic model for the active document.  By strongly holding onto it, we ensure
-    /// that it won't be GC'ed between feature requests from multiple features that care about it.  As the active
-    /// document has the most features running on it continuously, we definitely do not want to drop this.  Note: this
-    /// cached value is only to help with performance.  Not with correctness.  Importantly, the concept of 'active
-    /// document' is itself fundamentally racy.  That's ok though as we simply want to settle on these semantic models
-    /// settling into a stable state over time.  We don't need to be perfect about it.  They are intentionally not
-    /// locked either as we would only have contention right when switching to a new active document, and we would still
-    /// latch onto the new document very quickly.
+    /// Strongly held reference to the semantic models for the active document (and its related documents linked into
+    /// other projects).  By strongly holding onto then, we ensure that they won't be GC'ed between feature requests
```
Comment (ToddGrun) [inline src/Workspaces/Core/Portable/Workspace/Solution/Solution_SemanticModelCaching.cs:14]:
```suggestion
    /// other projects).  By strongly holding onto them, we ensure that they won't be GC'ed between feature requests
```

**Review by ToddGrun — APPROVED** [review]
:shipit:

**Comment by CyrusNajmabadi** [comment]
> I get an error in ServiceHubServicesTests compiling these changes locally

Have you tried turning it off and on again?

**Pull request merged**